release.sh

```diff
@@ -0,0 +1,181 @@
+      ;;
+  esac
+done
+
+# Prepare your local development environment by committing or stashing your changes. Work at the tip of main.
+echo "These are the pending changes in this branch"
+echo
+git diff
+
+echo "You should either commit[c]/stash[s] the changes: "
+l=1
+while [ $l -eq 1 ]; do
+  read -r commit_or_stash
+  case $commit_or_stash in
+    c)
+      git add && git commit && git push origin main
+      echo "Committed the changes on Development Branch"
+      l=0
+      ;;
+    s)
+      git stash
+      echo "Stashed the changes"
+      l=0
+      ;;
+    e)
+      exit
+      ;;
+    *)
+      echo "Unknown command. Try Again ..."
+      echo "You should either commit[c]/stash[s] the changes or input e to exit: "
+      ;;
+  esac
+done
+
+#Create a branch for the release: git checkout -b <release branch name>.
+echo "Enter the name of release branch: "
+read -r rBranch
+git checkout -b "$rBranch"
+#In a separate folder, create a fresh environment and activate it.
+cd "$BASE_DIR" || (echo "cd Failed" && exit )
+mkdir .release
+cd ".release" || (echo "cd Failed" && exit )
+python3 -m venv ternenv
+source ternenv/bin/activate
+#Clone the tern/main repository and cd into it.
+git clone --single-branch git@github.com:tern-tools/tern.git
+cd "tern" || (echo "cd Failed" && exit )
+
+########### PART 2 : Update Direct Dependencies and run tests ################
+
+#PART 2a : Update Direct Dependencies
+#Update direct dependencies and run tests.
+pip install wheel pip-tools twine
+pip-compile --upgrade --output-file upgrade.txt
+cd "$DIR" || (echo "cd Failed" && exit )
+# Some Manual work
+echo "Compare the module versions in upgrade.txt with requirements.txt in the development environment. Bump up versions if needed."
+echo "Press Y when you are done updating and want to continue? [y/n]: "
+l=1
+while [ $l -eq 1 ]; do
+  read -r manualWork1
+  case $manualWork1 in
+    n|N)
+      exit
+      ;;
+    y|Y)
+      echo "Continuing . . ."
+      l=0
+      ;;
+    *)
+      echo "Unknown command. Try Again:- "
+      ;;
+  esac
+done
+
+#Part 2b :Run Tests
+cd "$BASE_DIR/.release/ternenv/tern" || (echo "cd Failed" && exit )
+pip install . #Install tern
+tox #The tests
+echo "Which release version is this (eg 2.0.1)? : "
+read -r release_version
+pip-compile --generate-hashes --output-file "v$release_version-requirements.txt"
+cp "v$release_version-requirements.txt" "$DIR/docs/releases/"
+
+#################### Part 3: Write release notes. ################################
+cd "$DIR/docs/releases/" || (echo "cd Failed" && exit )
+echo "Choose among the 2 options:"
+echo "1. Write release notes all this stuff manually"
+echo "2. Enter the path to the release notes file"
+echo "Your choice [1/2]:- "
+l=1
+while [ $l -eq 1 ]; do
+  read -r release_notes_path
+  case $manualWork2 in
+    1)
+      touch "v$release_version.md"
+      echo "Press any key (or y) when you are done writing the release notes and want to continue; press n otherwise."
```

Comment:
Would it be possible to pass in a path to a file here? Usually release notes are written up in advance and copy/pasting them here risks losing some of the markdown formatting.

Reply:
@rnjudge This option has been kept dynamic and is asked for, by the script, during the script's execution. So during lines 118-122, if you want to use a path instead of manually-copy-pasting, you'll have to choose option 2. I can also completely remove this choice if you want.